### PR TITLE
Development

### PR DESCRIPTION
**Pull request by Shredder121**
- [X] caching for earlier repo fetches
- [X] make the lambda smaller
- [x] process all PRs for a given repo at startup (`startupRepo`)

### CODE REVIEW

**Comment by Shredder121** [comment]
Maybe also allow multiple repositories at startup?

**Comment by Shredder121** [comment]
@johnktims Do we still want persistence?
In my opinion, reading it all in at startup and work from there is sufficient.
Do you see something I don't?

File: src/main/java/com/querydsl/webhooks/GithubReviewWindow.java [hunk 1]
```diff
@@ -60,65 +65,55 @@
...
     public static void main(String... args) {
         GHEventApiServer.start(GithubReviewWindow.class, args);
     }
 
-    @Bean
-    public PullRequestHandler reviewWindowHandler(Environment environment) {
-        Duration defaultReviewWindow = Duration.parse(environment.getRequiredProperty("duration")); //duration is the default window
-        Map<String, ScheduledFuture<?>> asyncTasks = Maps.newConcurrentMap();
+    @Autowired
+    private GitHub gitHub;
```
Comment (johnktims) [inline src/main/java/com/querydsl/webhooks/GithubReviewWindow.java:81]: Nice!

Comment (johnktims) [inline src/main/java/com/querydsl/webhooks/GithubReviewWindow.java:81]: What do you think about making a spring boot starter for that project?

Comment (Shredder121) [inline src/main/java/com/querydsl/webhooks/GithubReviewWindow.java:81]: It's not _really_ necessary, but can be explored.
It already has its own well-known configuration (put it in the env, or in `~/.github`), so getting it up and running is already easy.
Getting it up and running easily in a Spring Boot context doesn't really seem like much added benefit in my opinion.

Comment (johnktims) [inline src/main/java/com/querydsl/webhooks/GithubReviewWindow.java:81]: Fair enough.

**Pull request merged**